odoo/addons/base/tests/test_mail.py

```diff
@@ -558,7 +565,34 @@ def send_message(this, message, smtp_from, smtp_to_list,
         msg['Message-Id'] = '<929227342217024.1596730490.324691772460938-example-30661-some.reference@test-123.example.com>'
         msg['References'] = '<345227342212345.1596730777.324691772483620-example-30453-other.reference@test-123.example.com>'
 
-        smtp = FakeSMTP()
-        self.patch(threading.currentThread(), 'testing', False)
-        self.env['ir.mail_server'].send_email(msg, smtp_session=smtp)
-        self.assertTrue(smtp.email_sent)
+        msg_on_the_wire = self.send_email(msg)
+
+        self.assertRegex(msg_on_the_wire, textwrap.dedent(r"""
+            From: .+? <joe@example\.com>
+            To: .+? <joe@example\.com>
+            Message-Id: <[0-9a-z.-]+@[0-9a-z.-]+>
+            References: (<[0-9a-z.-]+@[0-9a-z.-]+>\s*)+
+
+        """[1:]).replace('\n', '\r\n'))
```

Comment:
Using the regex here looks more confusing than it should be, an assert Equal should work and give a better error message if the test fails:
```suggestion
        self.assertEqual(msg_on_the_wire.strip(), textwrap.dedent(f"""
            From: =?utf-8?q?Jo=C3=A9?= Doe <joe@example.com>
            To: =?utf-8?q?Jo=C3=A9?= Doe <joe@example.com>
            Message-Id: {msg['Message-Id']}
            References: {msg['References']}
        """).replace('\n', '\r\n').strip())
```
(Feel free to adapt to remove the strips)